### ISSUE TITLE
Make all model classes torch modules with forward functions

### ISSUE THREAD

**Issue by Jubeku** [ecmwf/WeatherGenerator#1038]
### Is your feature request related to a problem? Please describe.

Not all classes in [engines.py](https://github.com/ecmwf/WeatherGenerator/blob/develop/src/weathergen/model/engines.py) inherit from `torch.nn.Module` with forward function which can be problematic as we saw we had non-zero dropout during validation. 

### Describe the solution you'd like

Make all classes inherit from `torch.nn.Module`.

### Organisation

ECMWF
